singularity-eos/eos/eos_vinet.hpp

```diff
@@ -0,0 +1,378 @@
+  _VIP[0] = robust::ratio(_B0, _rho0) +
+            robust::ratio(_A0 * _A0 * _B0 * _B0 * _T0,
+                          _rho0 * _rho0 * _Cv0); // sound speed squared
+  _VIP[1] = 3.0 / 2.0 * (_BP0 - 1.0);            // exponent eta0
+  // initializing with pressure coeffs to get energy coeffs into VIP
+  _VIP[2] = 1.0;                                                // prefactor d0
+  _VIP[3] = 0.0;                                                // prefactor d1
+  for (int ind = 4; ind < VinetInternalParametersSize; ind++) { //_d2tod40[0]=d2
+    _VIP[ind] = _d2tod40[ind - 4];                              // dn is in VIP[n+2]
+  }
+  for (int ind = VinetInternalParametersSize - 2; ind >= 2;
+       ind--) { // _VIP[42]=d40=f40 given,first calculated is _VIP[41]=f39
+    _VIP[ind] = _VIP[ind] - (ind) / _VIP[1] * _VIP[ind + 1]; // prefactors f40 to f0
+  }                                                          // _VIP[n+2]=fn, ind=n+2
+}
+
+PORTABLE_INLINE_FUNCTION void Vinet::Vinet_F_DT_func(const Real rho, const Real T,
+                                                     Real *output) const {
+  constexpr int pref0vp = -2, pref0vip = 2, maxind = VinetInternalParametersSize - 3;
+  Real sumP = 0.0, sumB = 0.0, sumE = 0.0;
+
+  Real x = std::cbrt(robust::ratio(_rho0, rho)); /*rho-dependent*/
+  Real x2inv = robust::ratio(1.0, x * x);
+  Real onemx = 1.0 - x;
+  Real etatimes1mx = _VIP[1] * onemx;
+  Real expetatimes1mx = exp(etatimes1mx);
+
+#pragma unroll
+  for (int ind = maxind; ind >= 2; ind--) {              //_d2tod40[0]=d2
+    sumP = _d2tod40[pref0vp + ind] + onemx * sumP;       //_d2tod40[38]=d40
+    sumB = _d2tod40[pref0vp + ind] * ind + onemx * sumB; //_d2tod40[-2+40]=d40
+    sumE = _VIP[pref0vip + ind] + onemx * sumE;          //_VIP[42]=f40
+  }                                                      //_VIP[2]=f0
+#pragma unroll
+  for (int ind = 1; ind >= 0; ind--) {
+    sumE = _VIP[pref0vip + ind] + onemx * sumE;
+  }
+  sumP = 1.0 + sumP * (onemx * onemx);
+  sumB = sumB * onemx;
+  sumE = _VIP[1] * onemx * sumE;
+
+  /* T0 isotherm */
+  Real energy = 9.0 * robust::ratio(_B0, _VIP[1] * _VIP[1] * _rho0) *
+                    (_VIP[pref0vip] - expetatimes1mx * (_VIP[pref0vip] - sumE)) -
+                (_A0 * _B0) * (robust::ratio(_T0, _rho0) - robust::ratio(_T0, rho));
+  Real pressure = 3.0 * _B0 * x2inv * onemx * expetatimes1mx * sumP;
+  Real temp = (1.0 + onemx * (_VIP[1] * x + 1.0)) * sumP + x * onemx * sumB;
+  temp = robust::ratio(_B0, _rho0) * x * expetatimes1mx * temp;
+
+  /* Go to required temperature */
+  energy = energy + _Cv0 * (T - _T0) + _E0;
+  pressure = pressure + (_A0 * _B0) * (T - _T0);
+  Real dpdrho = temp;
+  Real dpdt = _A0 * _B0;
+  Real dedt = _Cv0;
+  Real dedrho = robust::ratio(pressure - T * (_A0 * _B0), rho * rho);
+  Real entropy;
+  if (T < 0.0) {
+#ifndef NDEBUG
+    //    printf("warning!! negative temperature");
+    PORTABLE_WARN("Negative temperature");
```

Comment:
```suggestion
    PORTABLE_WARN("Negative temperature input");
```
I don't think we need the comment. I also added the word "input" to differentiate that the user is responsible for the negative temperature (i.e. it's not an internal calculation going wrong).